### ISSUE TITLE
Slow continous queries cause DB meltdown, without any error or warning in log

### ISSUE THREAD

**Issue by XANi** [influxdata/influxdb#6588]
After adding a continous query to downsample my data (from 10s to 10m):

```
CREATE CONTINUOUS QUERY down10test ON "stats" BEGIN SELECT mean(value) AS value INTO "stats"."test".:MEASUREMENT FROM "stats"."default"./.*/ GROUP BY time(10m), * END
```

I get sudden spike of writes and influx can't keep up on continous query (with smaller subset of data query works exactly as intended):

First "step" at 10:40 is enabling writes (memory usage spike is mostly from riemann, not influx), second step at 11:00 is adding that continuous query. I've dropped query at 11:08 but it didn't recover up until 11:22 (no log message) . Log doesn't show any errors, or that even query is running:

![zaznaczenie_2016-05-10_002](https://cloud.githubusercontent.com/assets/147394/15142119/dfc23266-16a3-11e6-9489-a753169ab32e.png)

```
[http] 2016/05/10 10:59:23 127.0.0.1 - - [10/May/2016:10:59:23 +0200] GET /query?chunked=true&db=&epoch=ns&q=down10test++CREATE+CONTINUOUS+QUERY+down10test+ON+%22stats%22+BEGIN+SELECT+mean%28value%29+AS+value+INTO+%22stats%22.%22test%22.%3AMEASUREMENT+FROM+%22stats%22.%22default%22.%2F.%2A%2F+GROUP+BY+time%2810m%29%2C+%2A+END HTTP/1.1 400 150 - InfluxDBShell/0.12.2 7dbbfacf-168d-11e6-adcf-000000000000 578.068µs
[query] 2016/05/10 10:59:32 CREATE CONTINUOUS QUERY down10test ON "stats" BEGIN SELECT mean(value) AS value INTO "stats".test.:MEASUREMENT FROM "stats"."default"./.*/ GROUP BY time(10m), * END
[http] 2016/05/10 10:59:32 127.0.0.1 - - [10/May/2016:10:59:32 +0200] GET /query?chunked=true&db=&epoch=ns&q=CREATE+CONTINUOUS+QUERY+down10test+ON+%22stats%22+BEGIN+SELECT+mean%28value%29+AS+value+INTO+%22stats%22.%22test%22.%3AMEASUREMENT+FROM+%22stats%22.%22default%22.%2F.%2A%2F+GROUP+BY+time%2810m%29%2C+%2A+END HTTP/1.1 200 46 - InfluxDBShell/0.12.2 82e20702-168d-11e6-adfb-000000000000 18.344685ms
[continuous_querier] 2016/05/10 11:00:00 executing continuous query down10test (2016-05-10 10:50:00 +0200 CEST to 2016-05-10 11:00:00 +0200 CEST)
[query] 2016/05/10 11:00:00 SELECT mean(value) AS value INTO "stats".test.:MEASUREMENT FROM "stats"."default"./.*/ WHERE time >= '2016-05-10T08:50:00Z' AND time < '2016-05-10T09:00:00Z' GROUP BY time(10m), *
[tsm1wal] 2016/05/10 11:00:24 tsm1 WAL starting with 10485760 segment size
[tsm1wal] 2016/05/10 11:00:24 tsm1 WAL writing to /datafs/influxdb/wal/stats/test/500
[tsm1] 2016/05/10 11:00:24 /datafs/influxdb/data/stats/test/500 database index loaded in 1.397µs
[tsm1] 2016/05/10 11:03:19 beginning level 1 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 11:03:19 compacting level 1 group (0) /datafs/influxdb/data/stats/default/491/000000005-000000001.tsm (#0)
[tsm1] 2016/05/10 11:03:19 compacting level 1 group (0) /datafs/influxdb/data/stats/default/491/000000006-000000001.tsm (#1)
[tsm1] 2016/05/10 11:03:20 compacted level 1 group (0) into /datafs/influxdb/data/stats/default/491/000000006-000000002.tsm.tmp (#0)
[tsm1] 2016/05/10 11:03:20 compacted level 1 group 0 of 2 files into 1 files in 1.668373801s
[retention] 2016/05/10 11:07:38 retention policy shard deletion check commencing
[retention] 2016/05/10 11:07:38 deleted shard group 474 from database stats, retention policy test
[tsm1] 2016/05/10 11:08:21 beginning level 1 compaction of group 0, 2 TSM files
[tsm1] 2016/05/10 11:08:21 compacting level 1 group (0) /datafs/influxdb/data/stats/10m/465/000000001-000000001.tsm (#0)
[tsm1] 2016/05/10 11:08:21 compacting level 1 group (0) /datafs/influxdb/data/stats/10m/465/000000002-000000001.tsm (#1)
[http] 2016/05/10 11:08:23 127.0.0.1 - - [10/May/2016:11:08:23 +0200] GET /query?chunked=true&db=&epoch=ns&q=DROP+CONTINUOUS+QUERY+down10test%3B HTTP/1.1 400 92 - InfluxDBShell/0.12.2 bf9388fd-168e-11e6-b447-000000000000 489.949µs
[tsm1] 2016/05/10 11:08:24 compacted level 1 group (0) into /datafs/influxdb/data/stats/10m/465/000000002-000000002.tsm.tmp (#0)
[tsm1] 2016/05/10 11:08:24 compacted level 1 group 0 of 2 files into 1 files in 3.002064846s
[query] 2016/05/10 11:08:33 DROP CONTINUOUS QUERY down10test ON "stats"
[http] 2016/05/10 11:08:33 127.0.0.1 - - [10/May/2016:11:08:33 +0200] GET /query?chunked=true&db=&epoch=ns&q=DROP+CONTINUOUS+QUERY+down10test+on+%22stats%22 HTTP/1.1 200 46 - InfluxDBShell/0.12.2 c558df15-168e-11e6-b477-000000000000 18.141128ms
```

I have around 110k "endpoints" (measurement + tag set) from collectd at rate of about 10k. Basically a default setup in config, only thing that changed are data dirs, version 0.12.2 . Server barely drops a sweat with that rate (most of CPU usage is riemann instance), in fact I can do similar downsampling in Riemann (albeit at high memory cost as it  have to store all of it in memory) and feed it to DB.

I dont _really_ understand why it is write churn considering write rate from CQ should be order of magnitude lower than, nevertheless if it "can't keep up" there should be at least warning in logs for that

**Comment by ruttosan86** [comment]
Hi Xani,
I think you are facing the same issue I found (in https://github.com/influxdata/influxdb/issues/6568).
I investigated a little the issue, and it seems that the disk spike is due to very high write requests that are very small in size (something like 1000 write requests with average size 12).

**Comment by XANi** [comment]
@ruttosan86 That would check out, we have HDD on that server so that ~170 is just max it can do.

**Comment by ruttosan86** [comment]
@XANi yes that's expected from a HDD setup, but I believe the problem is that the write load doesn't need to be like that, in fact if you write the same aggregated data that the CQ computes "from the outside" (like I did in the test I reported in #6568) you'll see a normal load on the disk, no spikes or anything... 
For what I saw, it seems like that anytime the clause "INTO" is used, is possible to have this sub-optimal disk usage pattern. That said, I'm no expert so I can be totally wrong...

**Comment by jwilder** [comment]
Fixed by #6732
